docs/user/kafka/data.rst

```diff
@@ -23,9 +23,17 @@ Kafka Topic Configuration
 -------------------------
 
 The Kafka topic for a given SimpleFeatureType will be created when calling ``createSchema`` (if it doesn't already
-exist). GeoMesa exposes a few configuration options through data store parameters. For more advanced options,
-the topic should be created using standard Kafka tools, and then the existing topic should be specified by name
-for the SimpleFeatureType (as described above).
+exist). GeoMesa exposes a few configuration options through data store parameters. Additional options can
+be configured by setting the user data key ``kafka.topic.config`` before calling ``createSchema``:
+
+.. code-block:: java
+
+    SimpleFeatureType sft = ....;
+    sft.getUserData().put("kafka.topic.config", "cleanup.policy=compact\nretention.ms=86400000");
```

Comment:
Nice.  

What happens if the topic already exists?  Should the user get an alert/error that the configurations will not be applied?

Should the configurations be applied:)?

Reply:
good point - it only gets applied if the topic doesn't exist. checking the status of the topic is a bit complicated, i had to pull out and replicate some internal kafka code in the [unit test ](https://github.com/locationtech/geomesa/pull/2840/files#diff-9779188cb8fb6f56a315daa1572fc8d8ae6182cc0540a343a5c3c92c9a86f51fR918-R925), but i'd rather not replicate that in src/main. and we probably don't want to log warnings since the topic may already be configured correctly? i think it's kind of an edge case so i'm ok leaving it as-is.

Reply:
Yeah... I can see it both ways.  Enabling compactions is Very Important, so I won't want someone to think they are doing so with SFT config which is applied only if it is the topic is new.  

That said, there's no guarantee that a warning would be read, so taking my worldview would almost need to promote that case to an error.:)

I feel like these are the tough corner cases where shipping it as-is is the right call for 90% of things.:)

Reply:
i realized we do already have a warning: https://github.com/locationtech/geomesa/blob/main/geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/data/KafkaDataStore.scala#L228
maybe i'll just update it to mention configs too